### ISSUE TITLE
jshint dependency

### ISSUE THREAD

**Issue by lundyio** [spalger/gulp-jshint#131]
What is the thought process behind making jshint a peer dependency? Just curious, had to update a couple builds this morning that were failing due to the dependency not being automatically installed.

**Comment by binarykitchen** [comment]
same here, it is breaking

**Comment by hbetts** [comment]
@lundyio and @binarykitchen based on this commit ( https://github.com/spalger/gulp-jshint/commit/3e7ad8431ac92ea040f3b02d2714633875a209b5 ) it looks like the intent was to emphasize that any JSHint-compatible tool could be used with `gulp-jshint` just by installing that tool.

So, using `gulp-jshint`, you can install `jshint` if you want to use that linter, or `jsxhint` if you would prefer to use that linter. (Documented under options https://github.com/spalger/gulp-jshint#options )

Though, I think I would have preferred a dedicated `gulp-jsxhint` than the extra installation requirements for `gulp-jshint`.

**Comment by KKfo** [comment]
Same here, it tells me ```npm WARN EPEERINVALID gulp-jshint@2.0.0 requires a peer of jshint@2.x but none was installed.``` But jshint is already installed.

**Comment by thelostspore** [comment]
+1

**Comment by mircoc** [comment]
+1

**Comment by jkphl** [comment]
+1

**Comment by Preen** [comment]
+1

**Comment by Baires** [comment]
:+1:

**Comment by hbetts** [comment]
This might seem like a silly question, but what does everyone on this issue want?

I ask because the issue was filed as a question and not a request to make any change to this package.

The commit, 3e7ad8431ac92ea040f3b02d2714633875a209b5, does a pretty good job explaining the reasoning. Do people agree with the reasoning?

**Comment by jkphl** [comment]
@hbetts While the reasoning behind might be good, at least for me the change completely breaks everything. It simply doesn't work. I'm facing the same situation as @KKfo described. Neither a globally nor a locally installed jshint does satisfy gulp-jshint's dependency. Maybe I'm doing something wrong, maybe it's a problem with Node 5, all I can say is that it worked like a charm prior to this change. Now it doesn't anymore, and I don't see what I can do about.

**Comment by spalger** [comment]
**TLDR** jshint is a dependency of gulp-jshint, it is listed in the package.json. There is one version of jshint that would make everyone happy. The purpose of semver is to prevent new versions from breaking your builds, please use it.

---

The reason that I felt it was necessary to take a different approach with the jshint dependency is because people are regularly confused as to why gulp-jshint doesn't update it's jshint dependency (#46, #102, #122). Jshint has also demonstrated in the past that it is pretty liberal with semver (see [jshint 2.5](https://github.com/jshint/jshint/releases/tag/2.5.0)). Then jshint published a release candidate to npm as it's latest version so gulp-jshint users were unknowingly installing a broken/pre-release version of jshint.

I feel like peerDependencies (especially the way that they are handled by npm3) helps mitigate the pain of all of these issues by putting everyone in control of their own jshint.

It was clear that a good number of people are already using npm 3 with node 5, so in order to prevent their builds from breaking I bumped the version to 2.0. That is what major version numbers are for. If you are automatically installing major version bumps then you are asking for trouble.

As far as I can tell there is no issue installing peerDependencies, so please run the install command at the top of the readme if you are having a hard time getting gulp-jshint to install.

![image](https://cloud.githubusercontent.com/assets/1329312/11456280/3c33790e-964a-11e5-9aaa-770288ccf145.png)

**Comment by jkphl** [comment]
@spalger Thanks for the explanation. Just to give you a short feedback: While I did understand all this before, it was the
```
npm install --save-dev jshint gulp-jshint
```
(taken from your screenshot) that made everything work now. This installed `jshint 2.8.0` locally, obviously — although the very same version was already present globally (but didn't suffice gulp-jshint's needs). I think this is what was troubling people (including me).

Thanks for helping in the end, though. ;)

**Comment by Flobin** [comment]
I still have this issue, even after installing jshint:

    (projenv) MacBook-Pro:posts robin$ gulp  
    [21:57:21] Using gulpfile ~/learning/python/parrot/posts/static/posts/gulpfile.js
    [21:57:21] Starting 'clean'...
    [21:57:21] Finished 'clean' after 4.86 ms
    [21:57:21] Starting 'default'...
    [21:57:21] Starting 'styles'...
    [21:57:21] Starting 'scripts'...
    ERROR: Can't find config file: .jshintrc
    (projenv) MacBook-Pro:posts robin$ sudo npm install jshint --save-dev
    npm WARN gulp-jshint@2.0.0 requires a peer of jshint@2.x but none was installed.
    (projenv) MacBook-Pro:posts robin$

**Comment by Baires** [comment]
@Flobin 
`rm -rf node_modules` and `npm install`

**Comment by Flobin** [comment]
@Baires yep, I did that. It still tried to install 2.9.1-rc1, so when I removed node_modules, set it to 2.8.0, and reinstalled, it worked fine.

**Comment by Chris-May** [comment]
@Flobin Thanks for the tip to downgrade jshint to 2.8.0. That's what got it "working" for me too.

**Comment by rshah9** [comment]
`npm install --save-dev jshint@2.8.0` this worked for me.

**Comment by ihsanberahim** [comment]
Thx @rshah9 . that fix work for me.

**Comment by mauaunot** [comment]
I'm starting to work with Gulp and was stuck with this error, thank you!!!! :)

**Comment by dylanh724** [comment]
The above seems to be a workaround, not a solution, unfortunately.
